### DISCUSSION TITLE
How to handle line joints in gradient lines?

### DISCUSSION THREAD

**Discussion by lianzhao** [visgl/deck.gl#7372] [Q&A] [answered]
I want to draw a gradient line like this

<img width="542" alt="image" src="https://user-images.githubusercontent.com/536786/199762923-a7a29f5d-3bf5-4924-b68e-b9c483c02290.png">

Here's my code so far
```js
class MyExtension extends LayerExtension {
  public getShaders = () => {
    return {
      inject: {
        "fs:DECKGL_FILTER_COLOR": `
          color = vec4(1,0,0,-geometry.uv.x * 0.5);
        `,
      },
    };
  };
}


const linesLayer = new GeoJsonLayer({
      data: myLInes,
     // ...
      extensions: [new MyExtension({})],
    });
```

It works well for the straight lines, but looks wired on the line joints...

<img width="179" alt="image" src="https://user-images.githubusercontent.com/536786/199764067-152c9153-d053-4c62-9fd3-cb84ff6055f3.png">

How to handle the joints properly?

**Comment by AdriSolid** [comment, marked as answer]
An additive blending could work:
```javascript
parameters: {
   blendFunc: [0x0302, 0x0304], // GL.SRC_ALPHA, GL.DST_ALPHA
   blendEquation: 0x8006 // GL.FUNC_ADD
}
```
So you get something like:
![Screenshot from 2022-11-04 13-32-22](https://user-images.githubusercontent.com/25487037/199973498-d5759ac8-bae9-4adb-8a1d-1d84fa7fa37b.png)

**Reply by lianzhao** [reply]
Thank you! Blending works for me!

**Comment by sorodrigo** [comment]
Is there a way to achieve this gradient blur in both sides of the line? Currently it only displays on the right side

**Reply by sorodrigo** [reply]
I found a way to do that:

<img width="411" alt="Screenshot 2024-02-15 at 17 15 12" src="https://github.com/visgl/deck.gl/assets/12124839/c46da15f-b39b-4bc5-b7fb-dda369bff315">


The changes I did were:
```glsl
class GlowExtension extends LayerExtension {
  getShaders() {
    return {
      inject: {
        'fs:DECKGL_FILTER_COLOR': `
          // Use the distance to the center to create a glow effect
          // The 0.5 is the center of the geometry cause it goes from 0 to 1
          // we want to make the glow effect stronger the closer we are to the center so we use abs(x - 0.5)
          float distanceToCenter = abs(geometry.uv.x - 0.5);

          // and we want a bigger area than just the center so we exclude the range from 0.25 to 0.75
          float alpha = distanceToCenter < 0.25 ? 1.0 : 1.0 - distanceToCenter;

          // Set the color with red and use the calculated alpha
          color = vec4(color.r, color.g, color.b, alpha);
      `,
      },
    };
  }
}
```
